File: src/main/java/infobip/test/shortener/account/RegisterUrlCommand.java
package infobip.test.shortener.account;

import infobip.test.shortener.model.UrlData;
import infobip.test.shortener.model.User;
import org.axonframework.commandhandling.annotation.TargetAggregateIdentifier;
import org.immutables.value.Value;

@Value.Immutable
public interface RegisterUrlCommand {

    @TargetAggregateIdentifier
    @Value.Derived
    default String getAccountId(){
        return getUser().getAccountId();
    }

    User getUser();
    UrlData getData();
    String getPath();

}

File: src/main/java/infobip/test/shortener/account/RegisterUrlCommand.kt
package infobip.test.shortener.account

import infobip.test.shortener.model.UrlData
import infobip.test.shortener.model.User
import org.axonframework.commandhandling.annotation.TargetAggregateIdentifier

data class RegisterUrlCommand(val user: User? = null,
                              val data: UrlData? = null,
                              val path: String? = null) {

    @TargetAggregateIdentifier
    fun getAccountId() = user?.accountId
}